src/js/reducers/conversation-reducer.js

```diff
@@ -25,16 +25,9 @@ const sortMessages = (messages) => messages.sort((messageA, messageB) => {
 });
 
 const addMessage = (messages, message) => {
-    const existingMessage = messages.find((m) => isEqual(m, message));
```

Comment:
I guess this, and the associated methods, became useless after the check for `source.id` ?